src/test/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMRevisionTest.java

```diff
@@ -0,0 +1,289 @@
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+
+public class PullRequestSCMRevisionTest {
+    /**
+     * All tests in this class only use Jenkins for the extensions
+     */
+    @ClassRule
+    public static JenkinsRule r = new JenkinsRule();
+
+
+    public static WireMockRuleFactory factory = new WireMockRuleFactory();
+
+    @Rule
+    public WireMockRule githubRaw = factory.getRule(WireMockConfiguration.options()
+            .dynamicPort()
+            .usingFilesUnderClasspath("raw")
+    );
+    @Rule
+    public WireMockRule githubApi = factory.getRule(WireMockConfiguration.options()
+            .dynamicPort()
+            .usingFilesUnderClasspath("api")
+            .extensions(
+                    new ResponseTransformer() {
+                        @Override
+                        public Response transform(Request request, Response response, FileSource files,
+                                                  Parameters parameters) {
+                            if ("application/json"
+                                    .equals(response.getHeaders().getContentTypeHeader().mimeTypePart())) {
+                                return Response.Builder.like(response)
+                                        .but()
+                                        .body(response.getBodyAsString()
+                                                .replace("https://api.github.com/",
+                                                        "http://localhost:" + githubApi.port() + "/")
+                                                .replace("https://raw.githubusercontent.com/",
+                                                        "http://localhost:" + githubRaw.port() + "/")
+                                        )
+                                        .build();
+                            }
+                            return response;
+                        }
+
+                        @Override
+                        public String getName() {
+                            return "url-rewrite";
+                        }
+
+                    })
+    );
+    private GitHub github;
+    private GHRepository repo;
+
+
+    @Before
+    public void prepareMockGitHub() throws Exception {
+        new File("src/test/resources/api/mappings").mkdirs();
+        new File("src/test/resources/api/__files").mkdirs();
+        new File("src/test/resources/raw/mappings").mkdirs();
+        new File("src/test/resources/raw/__files").mkdirs();
+        githubApi.enableRecordMappings(new SingleRootFileSource("src/test/resources/api/mappings"),
```

Comment:
Shouldn't you disable the recording after you've done it once? And then just checkin whatever was recorded?

Reply:
This is boilerplate as far as this change is concerned. 
I'm not innovating or fixing this here.